react/components/EXPERIMENTAL_FilterBar/FilterTag.js

```diff
@@ -0,0 +1,282 @@
+/* eslint-disable */
+import React, { PureComponent, Fragment } from 'react'
+import PropTypes from 'prop-types'
+import merge from 'lodash/merge'
+
+import IconClear from '../icon/Clear'
+import IconClose from '../icon/Close'
+import IconCaretDown from '../icon/CaretDown'
+import Statement from './UncoupledStatement'
+import Menu from './Menu'
+
+const OPEN_MENU_STYLE = { backgroundColor: '#dbe9fd' }
+const emptyVirtualStatement = {
+  subject: null,
+  verb: null,
+  object: null,
+  error: null,
+}
+
+const filterStatementBySubject = (statements = [], sbj, options = {}) => {
+  const hasStatement = statements.some(st => st.subject === sbj)
+  return hasStatement
+    ? statements.filter(st => st.subject === sbj)
+    : [
+        {
+          ...emptyVirtualStatement,
+          subject: sbj,
+          verb:
+            options[sbj] && options[sbj].verbs.length === 1
+              ? options[sbj].verbs[0].value
+              : null,
+        },
+      ]
+}
+
+class FilterTag extends PureComponent {
+  constructor(props) {
+    super(props)
+    this.filterMenuContainer = React.createRef()
+
+    this.state = {
+      isMenuOpen: false,
+      virtualStatement: filterStatementBySubject(
+        props.statements,
+        props.subject,
+        props.options
+      )[0],
+    }
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (
+      nextProps.statements &&
+      nextProps.statements.length > 0 &&
+      nextProps.subject &&
+      nextProps.statements.some(st => st.subject === nextProps.subject)
+    ) {
+      const statement = filterStatementBySubject(
+        nextProps.statements,
+        nextProps.subject,
+        nextProps.options
+      )[0]
+      return {
+        isMenuOpen: prevState.isMenuOpen,
+        virtualStatement: merge({}, statement, prevState.virtualStatement),
+      }
+    }
+  }
+
+  openMenu = () => {
+    if (this.state.isMenuOpen) return
+
+    document.addEventListener('mousedown', this.handleClickOutside)
+    this.setState({
+      isMenuOpen: true,
+      virtualStatement: filterStatementBySubject(
+        this.props.statements,
+        this.props.subject,
+        this.props.options
+      )[0],
+    })
+  }
+
+  closeMenu = () => {
+    if (!this.state.isMenuOpen) return
+
+    document.removeEventListener('mousedown', this.handleClickOutside)
+    this.setState({
+      isMenuOpen: false,
+      virtualStatement: filterStatementBySubject(
+        [],
+        this.props.subject,
+        this.props.options
+      )[0],
+    })
+  }
+
+  handleClickOutside = e => {
+    if (
+      this.filterMenuContainer &&
+      this.filterMenuContainer.current &&
+      !this.filterMenuContainer.current.contains(e.target) &&
+      this.state.isMenuOpen
+    ) {
+      this.closeMenu()
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.state.isMenuOpen) {
+      this.closeMenu()
```

Comment:
Does calling a function that calls `setState` on `componentWillUnmount` trigger the error of trying to update an unmounted component?